core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -61,20 +62,30 @@ static HttpResponseWriter streaming() {
      * Creates a new HTTP response that delegates to the {@link HttpResponse} produced by the specified
      * {@link CompletionStage}. If the specified {@link CompletionStage} fails, the returned response will be
      * closed with the same cause as well.
+     *
+     * @param stage the {@link CompletionStage} which will produce the actual {@link HttpResponse}
      */
     static HttpResponse from(CompletionStage<? extends HttpResponse> stage) {
-        requireNonNull(stage, "stage");
         final DeferredHttpResponse res = new DeferredHttpResponse();
-        stage.handle((delegate, thrown) -> {
-            if (thrown != null) {
-                res.close(Exceptions.peel(thrown));
-            } else if (delegate == null) {
-                res.close(new NullPointerException("delegate stage produced a null response: " + stage));
-            } else {
-                res.delegate(delegate);
-            }
-            return null;
-        });
+        delegateWhenStageComplete(stage, res);
+        return res;
+    }
+
+    /**
+     * Creates a new HTTP response that delegates to the {@link HttpResponse} produced by the specified
+     * {@link CompletionStage}. If the specified {@link CompletionStage} fails, the returned response will be
+     * closed with the same cause as well.
+     *
+     * @param stage the {@link CompletionStage} which will produce the actual {@link HttpResponse}
+     * @param defaultSubscriberExecutor the {@link EventExecutor} which will be used when a user subscribes
+     *                                  the returned {@link HttpResponse} using {@link #subscribe(Subscriber)}
+     *                                  or {@link #subscribe(Subscriber, SubscriptionOption...)}.
+     */
+    static HttpResponse from(CompletionStage<? extends HttpResponse> stage,
+                             EventExecutor defaultSubscriberExecutor) {
```

Comment:
nit: `defaultSubscriberExecutor` -> `subscriberExecutor`? 
I think this is a specified executor than default. :-)

Reply:
Fixed. Thanks!